home/models.py

```diff
@@ -748,6 +749,61 @@ def save_revision(
             revision.save(update_fields=["content"])
         return revision
 
+    def clean(self):
+        result = super().clean()
+        errors = {}
+
+        # The WA title is needed for all templates to generate a name for the template
+        if self.is_whatsapp_template and not self.whatsapp_title:
+            errors.setdefault("whatsapp_title", []).append(
+                ValidationError("All WhatsApp templates need a title.")
+            )
+        # The variable check is only for templates
+        if self.is_whatsapp_template and len(self.whatsapp_body.raw_data) > 0:
+            whatsapp_message = self.whatsapp_body.raw_data[0]["value"]["message"]
+            vars_in_msg = re.findall(r"{{(.*?)}}", whatsapp_message)
+            non_digit_variables = [var for var in vars_in_msg if not var.isdecimal()]
+
+            if non_digit_variables:
+                errors.setdefault("whatsapp_body", []).append(
+                    StreamBlockValidationError(
+                        {
+                            0: StreamBlockValidationError(
+                                {
+                                    "message": ValidationError(
+                                        f"Please provide numeric variables only. You provided {non_digit_variables}."
+                                    )
+                                }
+                            )
+                        }
+                    )
+                )
+
+            # Check variable order
+            actual_digit_variables = [var for var in vars_in_msg if var.isdecimal()]
+            expected_variables = [
+                str(j + 1) for j in range(len(actual_digit_variables))
+            ]
+            if actual_digit_variables != expected_variables:
+                errors.setdefault("whatsapp_body", []).append(
+                    StreamBlockValidationError(
+                        {
+                            0: StreamBlockValidationError(
+                                {
+                                    "message": ValidationError(
+                                        f'Variables must be sequential, starting with "{{1}}". Your first variable was "{actual_digit_variables[0]}"'
```

Comment:
Is there a reason why we're only giving the value for the first variable? If I have a message with `{{1}} {{3}} {{2}}`, then the error message I get is `Variables must be sequential, starting with "{1}". Your first variable was "1"`, which isn't very useful. Maybe `Your variables are {actual_digit_variables}`?

Reply:
Not sure why I put that there, its been removed.